Pirat/Services/Resource/ResourceUpdateService.cs

```diff
@@ -132,24 +134,128 @@ public Task delete(string token)
             return Task.CompletedTask;
         }
 
-        public Task ChangeInformation(string token, Provider provider)
+        public Task<int> ChangeInformation(string token, Provider provider)
         {
-            return Task.CompletedTask;
+
+            AddressEntity location = new AddressEntity().build(provider.address);
+            _addressMaker.SetCoordinates(location);
+
+            var query = from o in _context.offer
+                join ap in _context.address on o.address_id equals ap.id
+                where o.token == token
+                select new {o, ap};
+
+            var l = query
+                .Select(x => x)
+                .ToList();
+            l.ForEach(collection =>
```

Comment:
If the token exists, there should be exactly one offer, right? Using ForEach suggests that there could be multiple offers.

Reply:
ForEach makes it easier to change the attributes in one expression and query is an enumerable + after this expression checks are made if too many rows are changed (which never should happen).